src/Internal/Service/Level3Service.php

```diff
@@ -0,0 +1,192 @@
+use WCPay\Internal\Service\OrderService;
+use WCPay\Exceptions\Order_Not_Found_Exception;
+
+/**
+ * Service for generating Level 3 data from orders.
+ */
+class Level3Service {
+	/**
+	 * Order service.
+	 *
+	 * @var OrderService
+	 */
+	private $order_service;
+
+	/**
+	 * WC_Payments_Account instance to get information about the account
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+
+	/**
+	 * Service constructor.
+	 *
+	 * @param OrderService        $order_service Order service.
+	 * @param WC_Payments_Account $account       WooPayments account.
+	 */
+	public function __construct(
+		OrderService $order_service,
+		WC_Payments_Account $account
+	) {
+		$this->order_service = $order_service;
+		$this->account       = $account;
+	}
+
+	/**
+	 * Create the level 3 data array to send to Stripe when making a purchase.
+	 *
+	 * @param int $order_id The order that is being paid for.
+	 * @return array        The level 3 data to send to the API.
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function get_data_from_order( int $order_id ): array {
+		$order = $this->order_service->get_order( $order_id );
+
+		$merchant_country = $this->account->get_account_country();
+		// We do not need to send level3 data if merchant account country is non-US.
+		if ( 'US' !== $merchant_country ) {
+			return [];
+		}
+
+		// Get the order items. Don't need their keys, only their values.
+		// Order item IDs are used as keys in the original order items array.
+		$order_items = array_values( $order->get_items( [ 'line_item', 'fee' ] ) );
+		$currency    = $order->get_currency();
+
+		$process_item  = function( $item ) use ( $currency ) {
+			return $this->process_item( $item, $currency );
+		};
+		$items_to_send = array_map( $process_item, $order_items );
+
+		if ( count( $items_to_send ) > 200 ) {
+			// If more than 200 items are present, bundle the last ones in a single item.
+			$items_to_send = array_merge(
+				array_slice( $items_to_send, 0, 199 ),
+				[ $this->bundle_level3_data_from_items( array_slice( $items_to_send, 200 ) ) ]
+			);
+		}
+
+		$level3_data = [
+			'merchant_reference' => (string) $order->get_id(), // An alphanumeric string of up to  characters in length. This unique value is assigned by the merchant to identify the order. Also known as an “Order ID”.
+			'customer_reference' => (string) $order->get_id(),
+			'shipping_amount'    => $this->prepare_amount( (float) $order->get_shipping_total() + (float) $order->get_shipping_tax(), $currency ), // The shipping cost, in cents, as a non-negative integer.
+			'line_items'         => $items_to_send,
+		];
+
+		// The customer’s U.S. shipping ZIP code.
+		$shipping_address_zip = $order->get_shipping_postcode();
+		if ( WC_Payments_Utils::is_valid_us_zip_code( $shipping_address_zip ) ) {
+			$level3_data['shipping_address_zip'] = $shipping_address_zip;
+		}
+
+		// The merchant’s U.S. shipping ZIP code.
+		$store_postcode = get_option( 'woocommerce_store_postcode' );
```

Comment:
Could we create GetOptionService or something similar that will mimic this behavior when we need it? It could help us in unit testing, and we could 'cache' this function to avoid trip to the database. Perhaps we can implement it in another PR if you are ok with it.

Reply:
In this case I simply moved the code to the service, and didn't focus on this, which is my mistake. For simple methods like this, we're currently using the `LegacyProxy` class ([docs](https://github.com/Automattic/woocommerce-payments/blob/rpp/extract-gateway-methods/src/README.md#legacy-proxy)), which I will do here shortly.